File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.cpp
#include <string>
#include <iostream>
using namespace std;

bool solution(string s)
{
    int p=0, y=0;
    
    for (auto i: s) {
        if (i == 'p' || i == 'P')
            p++;
        if (i == 'y' || i == 'Y')
            y++;
    }

    return p == y;
}

File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.js
function solution(s){
    let p = 0, y = 0;
    
    for (const c of s) {
        switch(c) {
            case 'p':
            case 'P': {
                p++;
                break;   
            }
            case 'y':
            case 'Y': {
                y++;
                break;
            }
            default:
                break;
        }
    }

    return (p == y);
}